src/main/java/org/apache/maven/plugins/jmod/JModCreateMojo.java

```diff
@@ -559,7 +577,12 @@ private Commandline createJModCreateCommandLine( File resultingJModFile )
             argsFile.println( moduleVersion );
         }
 
-        if ( !pathElements.isEmpty() )
+        if ( classPath != null && !classPath.trim().isEmpty() )
```

Comment:
You should be able to detect if there are classpath elements by looping over all the pathElements.

Reply:
Just one observation here: jmod create can receive both --module-path and --class-path. In this case, I think it is a better solution is to segregate JMOD artifacts and modular JARs from non-modular JARs.

I am using *Java 9 Modularity* book and the official Oracle documentation for references.